examples/toy_mixture_model.py

```diff
@@ -0,0 +1,163 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+ A toy mixture model to provide a simple example for implementing discrete enumeration.
+
+ (A) -> [B] -> (C)
+
+ A is an observed Bernoulli variable with Beta prior.
+ B is a hidden variable which is a mixture of two Bernoulli distributions (with Beta priors),
+ chosen by A being true or false.
+ C is observed, and like B, is a mixture of two Bernoulli distributions (with Beta priors),
+ chosen by B being true or false.
+ There is a plate over the three variables for n independent observations of data.
+
+ Because B is hidden and discrete we wish to marginalize it out of the model.
+ This is done by:
+    1) marking the model method with `@pyro.infer.config_enumerate`
+    2) marking the B sample site in the model with `infer={"enumerate": "parallel"}`
+    3) passing `pyro.infer.SVI` the `pyro.infer.TraceEnum_ELBO` loss function
+"""
+
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from torch.distributions import constraints
+from torch.distributions.bernoulli import Bernoulli
+from torch.distributions.beta import Beta
+from tqdm import tqdm
+
+import pyro
+import pyro.distributions as dist
+import pyro.infer
+import pyro.optim
+from pyro.ops.indexing import Vindex
+
+smoke_test = ('CI' in os.environ)
+
+
+def main():
```

Comment:
nit: could you maybe remove some of the extra blank lines throughout? this is really just a stylistics thing, but we don't use so much whitespace in the rest of the codebase...

Reply:
Done. Removed all the ones within methods. Left the double newlines between methods as per my PEP8 checker.